frontend/src/components/StockNews/StockNews.jsx

```diff
@@ -1,10 +1,62 @@
 import "./StockNews.css";
+import { useState, useEffect, useContext } from "react";
+import axios from "axios";
+import { formatDate, firstWord } from "../../utils.js";
+import { UserContext } from "../App/App";
 
-export default function StockNews() {
+export default function StockNews({ stocks }) {
+    const [stockNews, setStockNews] = useState([]);
+    const [currentStock, setCurrentStock] = useState(null);
+    const { setError } = useContext(UserContext);
+
```

Comment:
I noticed in your PR video there is a slight glitch(?) in the news articles rendering when switching between stocks. Not a blocker for now (consider a P1), might be worth double-checking if the dependencies list in useEffect or something else causing the component to re-render?